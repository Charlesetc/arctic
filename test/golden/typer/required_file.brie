
define example_function [
  define y 2
  < x = y >
  -- < x = 2 >
]

define main [
  -- technically this will be parsed as a test
  -- I guess that's good; it should be parseable.
  -- nothing cool though
  -- may as well call a function:
  (example_function ()).x
]
