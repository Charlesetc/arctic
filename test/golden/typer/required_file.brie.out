root[(ident(define):: ident(example_function):: block[][(ident(define):: ident(y):: ident(2)::)::, (<x = (ident(y)::)::>::)::]::Function)::, (ident(define):: ident(main):: block[][((ident(example_function)::Function ()::Unit)::<x = Integer>.x::Integer)::Integer]::Function)::]::